relation-extraction/src/main/java/org/cogcomp/re/RelationAnnotator.java

```diff
@@ -33,25 +33,36 @@
     private ACERelationConstrainedClassifier constrainedClassifier;
     private Gazetteers gazetteers;
     private WordNetManager wordNet;
-    private MentionAnnotator mentionAnnotator;
+    private String type = "ACE"; // default relation type
 
     public RelationAnnotator() {
         this(true);
     }
 
-
     public RelationAnnotator(boolean lazilyInitialize) {
-        super(ViewNames.MENTION, new String[]{ViewNames.POS, ViewNames.DEPENDENCY_STANFORD, ViewNames.SHALLOW_PARSE}, lazilyInitialize);
+        this(lazilyInitialize, "ACE");
+    }
+
+    public RelationAnnotator(boolean lazilyInitialize, String type) {
+        super(ViewNames.RELATION, new String[]{ViewNames.MENTION, ViewNames.POS, ViewNames.DEPENDENCY_STANFORD, ViewNames.SHALLOW_PARSE}, lazilyInitialize);
+        this.type = (type.contains("ACE")) ? "ACE" : "SEMEVAL";
     }
 
     @Override
     public void initialize(ResourceManager rm) {
         try {
-            mentionAnnotator = new MentionAnnotator("ACE_TYPE");
             Datastore ds = new Datastore(new ResourceConfigurator().getDefaultConfig());
-            File modelDir = ds.getDirectory("org.cogcomp.re", "ACE_GOLD_BI", 1.0, false);
-            String modelFile = modelDir.getPath() + File.separator + "ACE_GOLD_BI" + File.separator + "ACE_GOLD_BI.lc";
-            String lexFile = modelDir.getPath() + File.separator + "ACE_GOLD_BI" + File.separator + "ACE_GOLD_BI.lex";
+            String modelFile;
+            String lexFile;
+            if (type.equals("ACE")) {
+                File modelDir = ds.getDirectory("org.cogcomp.re", "ACE_GOLD_BI", 1.0, false);
+                modelFile = modelDir.getPath() + File.separator + "ACE_GOLD_BI" + File.separator + "ACE_GOLD_BI.lc";
+                lexFile = modelDir.getPath() + File.separator + "ACE_GOLD_BI" + File.separator + "ACE_GOLD_BI.lex";
+            } else {
+                File modelDir = ds.getDirectory("org.cogcomp.re", "SEMEVAL", 1.1, false);
+                modelFile = modelDir.getPath() + File.separator + "SEMEVAL" + File.separator + "SEMEVAL.lc";
+                lexFile = modelDir.getPath() + File.separator + "SEMEVAL" + File.separator + "SEMEVAL.lex";
+            }
```

Comment:
How about here?

Reply:
Can you change this line https://github.com/danyaljj/illinois-cogcomp-nlp-1/blob/60b45aadcd689fbe5fc6e0af66ca0aa585c0cff7/relation-extraction/src/main/java/org/cogcomp/re/RelationAnnotator.java#L169

to 

```Java
if (!tag.equals("NOT_RELATED") && !tag.equals("Other")) {
```

Because the null type in SemEval relations is called "Other"

Reply:
@Slash0BZ this is ok? I think it's not working ....

Reply:
Did you use the same classifiers or different classifiers?

Reply:
My bad, I forgot it's a different classifier. The classifier for SemEval is `semeval_relation_classifier`. Also you don't need the constrainedClassifier anymore, just directly use `semeval_relation_classifier` and then load model and lexicon